ui/src/app/machines/search.js

```diff
@@ -10,55 +10,39 @@ export const getCurrentFilters = search => {
   if (!search) {
     return filters;
   }
-  const terms = search.split(" ");
-  let processingFilter;
-  terms.forEach(term => {
-    if (processingFilter) {
-      const filter = filters[processingFilter];
-      // A previous term had a space in it. This may contain multiple comma
-      // separated terms, the first of which will be part of the last term to
-      // be added to the filter.
-      term.split(",").forEach((termItem, i) => {
-        termItem = termItem.replace(")", "");
-        if (i === 0) {
-          // If this is the first term it is part of the last term to be added.
-          filter[filter.length - 1] += ` ${termItem}`;
-        } else {
-          // This is a new term.
-          filter.push(termItem);
-        }
-      });
-      // If the term contains the ending ')' then it's the last in the group.
-      if (term.includes(")")) {
-        processingFilter = null;
+  // Match filters with parens e.g. 'status:(new,deployed)'.
+  // Then: match filters without parens e.g. 'status:new,deployed' or 'status'
+  const filterMatchingRegex = search.matchAll(/(\b\w+:!?\([^)]+\))|(\w+\S+)/g);
+  Array.from(filterMatchingRegex).forEach(([group]) => {
```

Comment:
You could use the spread operator to make this a bit shorter if you'd like.
```JavaScript
[...filterMatchingRegex]
```

Reply:
Done.